test/L2/paused/L2RewardPaused.t.sol

```diff
@@ -0,0 +1,283 @@
+
+        bridge = vm.addr(uint256(bytes32("bridge")));
+        remoteToken = vm.addr(uint256(bytes32("remoteToken")));
+
+        vm.prank(address(this), address(this));
+
+        l2LiskToken = new L2LiskToken(remoteToken);
+        l2LiskToken.initialize(bridge);
+
+        vm.stopPrank();
+
+        l2StakingImplementation = new L2Staking();
+        l2Staking = L2Staking(
+            address(
+                new ERC1967Proxy(
+                    address(l2StakingImplementation),
+                    abi.encodeWithSelector(l2Staking.initialize.selector, address(l2LiskToken))
+                )
+            )
+        );
+
+        l2LockingPositionImplementation = new L2LockingPosition();
+        l2LockingPosition = L2LockingPosition(
+            address(
+                new ERC1967Proxy(
+                    address(l2LockingPositionImplementation),
+                    abi.encodeWithSelector(l2LockingPosition.initialize.selector, address(l2Staking))
+                )
+            )
+        );
+
+        l2VotingPowerImplementation = new L2VotingPower();
+        l2VotingPower = L2VotingPower(
+            address(
+                new ERC1967Proxy(
+                    address(l2VotingPowerImplementation),
+                    abi.encodeWithSelector(l2VotingPower.initialize.selector, address(l2LockingPosition))
+                )
+            )
+        );
+
+        l2Staking.initializeLockingPosition(address(l2LockingPosition));
+        l2LockingPosition.initializeVotingPower(address(l2VotingPower));
+
+        l2RewardImplementation = new L2Reward();
+
+        vm.expectEmit(true, true, true, true);
+        emit L2Staking.LiskTokenContractAddressChanged(address(0x0), address(l2LiskToken));
+
+        l2Reward = L2Reward(
+            address(
+                new ERC1967Proxy(
+                    address(l2RewardImplementation),
+                    abi.encodeWithSelector(l2Reward.initialize.selector, address(l2LiskToken))
+                )
+            )
+        );
+
+        l2Reward.initializeLockingPosition(address(l2LockingPosition));
+
+        vm.expectEmit(true, true, true, true);
+        emit L2Reward.StakingContractAddressChanged(address(0x0), address(l2Staking));
+        l2Reward.initializeStaking(address(l2Staking));
+
+        assertInitParamsEq(l2Reward);
+
+        l2Staking.addCreator(address(l2Reward));
+
+        vm.startPrank(bridge);
+        l2LiskToken.mint(staker, convertLiskToSmallestDenomination(100));
+        vm.stopPrank();
+
+        // create a position to have it for testing different function calls inside Reward contract
+        vm.startPrank(staker);
+        l2LiskToken.approve(address(l2Reward), convertLiskToSmallestDenomination(100));
+        ID = l2Reward.createPosition(convertLiskToSmallestDenomination(100), 150);
+        vm.stopPrank();
+
+        stakerPositions = new uint256[](1);
+        stakerPositions[0] = ID;
+
+        upgradeLockingPositionContractToPausedVersion();
```

Comment:
Shouldn't the pausing of the rewards contract be technically independent of the lockingPosition contract?

Reply:
Before implementing full paused version of Reward contract, we need to pause LockingPosition that Reward contract was fully paused.
Currently this is not anymore the case, so it may be removed.

Done.